### DISCUSSION TITLE
FIR filter adjustment of DeMask

### DISCUSSION THREAD

**Discussion by PMX-BradChen** [asteroid-team/asteroid#515] [Q&A]
Hi Sir,

You did a great job in the demask field. We also study the same field and get some problems on how to adjust the fir filter coefficients to meet the real condition. Could you help to share related experiences on filter adjustment by hand?

Thanks,
Brad

**Comment by mpariente** [comment]
It's mainly @popcornell who did this, but he used a scipy function for linear filtering interpolation IIRC. 
Samu, do you have the name of the function please?

**Reply by popcornell** [reply]
Hi Brad,
Sorry for my late response.

I used firwin2 and the mag response for each mask type was taken from a paper linked in the README in demask recipe. 

Noise was applied to the points in the magnitude response in order to vary it a bit. 

Unfortunately generalization is quite challenging. You could try to add more noise and reverberation to the dataset.

**Reply by PMX-BradChen** [reply]
Hi Samu and Mpariente, 

Thanks for your reply and info. It helps a lot in corpus generation. Regarding mag response, are the filter coefficients the same between demask_dataset.py and ref [1]? The filter adjustment by hand is meaning that applied magnitude response with point variation in firwin2. Then applied noise and reverberation to the dataset after firwin2.

Thanks,
Brad

**Reply by PMX-BradChen** [reply]
Hi Samuele and Pariente,

We want to investigate the demask performance of the higher sampling rate IO. Did you have Impulse response coefficients based on a 32k sampling rate for reference?

Thanks,
Brad

**Reply by popcornell** [reply]
You can probably derive them from Corey, Ryan M., Uriah Jones, and Andrew C. Singer. "Acoustic effects of medical, cloth, and transparent face masks on speech signals." arXiv preprint arXiv:2008.04521 (2020). They report mag response up to 16 kHz.
